samcli/lib/runtimes/base.py

```diff
@@ -0,0 +1,342 @@
+import logging
+import os
+from dataclasses import dataclass, field
+from enum import Enum, unique
+from pathlib import Path
+from typing import Dict, List, Optional, Tuple, cast
+
+LOG = logging.getLogger(__name__)
+
+_init_path = Path(os.path.dirname(__file__)).parent.parent
+_templates = _init_path / "lib" / "init" / "templates"
+_lambda_images_templates = _init_path / "lib" / "init" / "image_templates"
+
+
+@unique
+class Architecture(Enum):
+    X86_64 = "x86_64"
+    ARM64 = "arm64"
+
+
+@dataclass
+class FamilyDataMixin:
+    key: str
+    layer_subfolder: Optional[str]
+    dep_manager: List[Tuple[str, Path]] = field(default_factory=list)  # (package manager, path to local init template)
+    build: bool = True
+    eb_code_binding: Optional[str] = None  # possible values are "Java8", "Python36", "Go1", "TypeScript3"
+    include_in_runtime_dep_template_mapping: bool = field(repr=False, default=True)
+
+
+@unique
+class Family(FamilyDataMixin, Enum):
+    DOTNET = (
+        "dotnet",
+        "dotnet",
+        [("cli-package", _templates / "cookiecutter-aws-sam-hello-dotnet")],
+        True,
+    )
+    GO = "go", None, [("mod", _templates / "cookiecutter-aws-sam-hello-golang")], False, "Go1"
+    JAVA = (
+        "java",
+        "java",
+        [
+            ("maven", _templates / "cookiecutter-aws-sam-hello-java-maven"),
+            ("gradle", _templates / "cookiecutter-aws-sam-hello-java-gradle"),
+        ],
+        True,
+        "Java8",
+    )
+    NODEJS = (
+        "nodejs",
+        "nodejs",
+        [("npm", _templates / "cookiecutter-aws-sam-hello-nodejs")],
+        True,
+    )
+    PROVIDED = "provided", "", [], False, None, False  # type: ignore [var-annotated]
+    PYTHON = (
+        "python",
+        "python",
+        [("pip", _templates / "cookiecutter-aws-sam-hello-python")],
+        True,
+        "Python36",
+    )
+    RUBY = "ruby", "ruby/lib", [("bundler", _templates / "cookiecutter-aws-sam-hello-ruby")]
+
+
+@dataclass
+class RuntimeDataMixin:
+    """
+    Definition of a SAM Runtime
+    """
+
+    key: str
+    family: Family
+    lambda_image: Optional[str] = None
+    archs: List[Architecture] = field(default_factory=list)
+    is_lambda_enum: bool = True
+    # build_workflow: WorkflowConfig
+    # workflow_selector: ???
+
+    # debug settings - samcli/local/docker/lambda_debug_settings.py
+    # debug_entry_setting: callable
+
+    # runtime enum - samcli/local/docker/lambda_image.py
+
+    # DEPRECATED_RUNTIMES - samcli/lib/build/constants.py
+    is_deprecated: bool = False
+
+    # local manifest - samcli/lib/init/local_manifest.json
+    # is local manifest useful at all?
```

Comment:
I believe we fall back to the local manifest if there's an error fetching from GitHub

Reply:
yeah, I think my conclusion was that we can generate it during runtime as a dictionary instead of reading from a json file. That's why I have the `to_local_init_manifest` method to be implemented below

Reply:
Do you have a strong reason for generating it instead of reading from a file?

Reply:
Having a file means we need to have another location to maintain the list of runtimes. The goal for runtime improvement is to minimize the number of changes needed for adding/deprecating a runtime